src/Microsoft.IdentityModel.Tokens/SymmetricSignatureProvider.cs

```diff
@@ -78,7 +78,7 @@ public SymmetricSignatureProvider(SecurityKey key, string algorithm)
             }
 
             if (_keyedHash == null)
-                throw LogHelper.LogExceptionMessage(new ArgumentOutOfRangeException(nameof(key), LogHelper.FormatInvariant(LogMessages.IDX10641, key)));
+                throw LogHelper.LogExceptionMessage(new NotSupportedException(LogHelper.FormatInvariant(LogMessages.IDX10641, key)));
```

Comment:
I think we should throw InvalidOperationException here. Change message to state that "GetKeyedHasAlgorithm returned null, key: {0}, algorithm {1}."